tests/components/envisalink/test_alarm_control_panel.py

```diff
@@ -0,0 +1,21 @@
+"""Tests for envisalink alarm control panel."""
+
+import pytest
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import slugify
+
+from .common import PARTITIONS, setup_platform
+
+
+@pytest.mark.parametrize("expected_lingering_tasks", [True])
+async def test_alarm_control_panel_entity_registry(hass: HomeAssistant) -> None:
+    """Test the sensors are registered in entity registry."""
+    await setup_platform(hass)
+    entity_registry = er.async_get(hass)
```

Comment:
Please take a look at  #90782
You could do something similar here, extracting all registered entities and using snapshot assertion.

Reply:
It seems that Snapshots are more oriented to reproducability of numerous state variables as opposed to the correctness of the state. It assumes the fitst snapshot is correct.

Reply:
OK I can eliminate the for loop .... but I still do not know if the expected registry entries are correct only that the results are repeatable.